packages/devextreme/js/__internal/scheduler/appointment_popup/appointment_popup.test.ts

```diff
@@ -422,51 +416,387 @@ describe('Appointment Popup Form', () => {
+      const appointment = {
+        text: 'Test Appointment',
+        startDate: new Date(2017, 4, 1, 9, 30),
+        endDate: new Date(2017, 4, 1, 11),
+      };
+
+      const { POM, scheduler } = await createScheduler(getDefaultConfig());
+
+      scheduler.showAppointmentPopup(appointment);
+
+      const recurrenceGroup = $(POM.popup.element).find('.dx-scheduler-form-recurrence-group');
+
+      expect(recurrenceGroup.hasClass('dx-scheduler-form-recurrence-hidden')).toBe(true);
+
+      // @ts-expect-error
+      const repeatEditor = $(POM.popup.repeatEditor).dxSelectBox('instance');
+      repeatEditor.option('value', 'weekly');
+
+      expect(recurrenceGroup.hasClass('dx-scheduler-form-recurrence-hidden')).toBe(false);
+    });
+
+    it('Should discard changes when canceling from recurrence form', async () => {
+      const data = [
+        {
+          text: 'meet',
+          startDate: new Date(2017, 4, 22, 9, 30),
+          endDate: new Date(2017, 4, 22, 11, 30),
+        },
+      ];
+
+      const { POM, scheduler } = await createScheduler({
+        dataSource: data,
+        currentDate: new Date(2017, 4, 22),
+      });
+
+      POM.openPopupByDblClick('meet');
+
+      // @ts-expect-error
+      const repeatEditor = $(POM.popup.repeatEditor).dxSelectBox('instance');
+      repeatEditor.option('value', 'weekly');
+
+      scheduler.hideAppointmentPopup(true);
+
+      POM.openPopupByDblClick('meet');
+
+      POM.popup.getEditSeriesButton().click();
+
+      // @ts-expect-error
+      const repeatEditor2 = $(POM.popup.repeatEditor).dxSelectBox('instance');
+      repeatEditor2.option('value', 'daily');
+
+      POM.popup.getBackButton().click();
+
+      scheduler.hideAppointmentPopup();
+
+      POM.openPopupByDblClick('meet');
+
+      POM.popup.getEditSeriesButton().click();
+
+      // @ts-expect-error
+      const repeatEditor3 = $(POM.popup.repeatEditor).dxSelectBox('instance');
+      expect(repeatEditor3.option('value')).toBe('weekly');
+    });
+
+    describe('Recurrence frequency types', () => {
+      it('Should show day selection buttons when frequency is weekly', async () => {
+        const appointment = {
+          text: 'Test Appointment',
+          startDate: new Date(2017, 4, 1, 9, 30),
+          endDate: new Date(2017, 4, 1, 11),
+        };
+
+        const { POM, scheduler } = await createScheduler(getDefaultConfig());
+
+        scheduler.showAppointmentPopup(appointment);
+
+        // @ts-expect-error
+        const repeatEditor = $(POM.popup.repeatEditor).dxSelectBox('instance');
+        repeatEditor.option('value', 'weekly');
+
+        const buttons = repeatEditor.option('buttons');
+        const settingsButton = buttons?.find((btn) => btn.name === 'settings');
+        if (settingsButton?.options?.onClick) {
+          settingsButton.options.onClick();
+        }
+
+        const dayButtons = $(POM.popup.element).find('.dx-scheduler-recurrence-byday-buttons');
```

Comment:
Can you add this to popup's POM? Can you please do so for all other raw selectors that you've added?